config/routes.rb

```diff
@@ -1,6 +1,34 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  mount Rswag::Api::Engine => '/api-docs'
+  mount Rswag::Ui::Engine => '/api-docs'
+  devise_for :users, skip: [:sessions]
+  
+  as :user do
+    get 'login', to: 'devise/sessions#new', as: :new_user_session
+    post 'login', to: 'devise/sessions#create', as: :user_session
+    delete 'logout', to: 'devise/sessions#destroy', as: :destroy_user_session
+  end
+  
+  root 'users#index'
+  
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show, :new, :create] do
+      resources :likes, only: [:create, :destroy]
+      resource :deletion, only: [:destroy], controller: 'post_deletions'
+      resources :comments, only: [:new, :create, :show] do
+        resource :deletion, only: [:destroy], controller: 'comment_deletions'
+      end
+      
+    end
+  end
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  namespace :api do
+    namespace :v1 do
+      resources :users, only: [] do
+        resources :posts, only: [:index] do
+          resources :comments, only: [:index, :create]
+        end
+      end
```

Comment:
- [ ] As I mentioned in the previous review and as required for the project all API your endpoints to receive JSON and responds in a JSON format as well. Kindly ensure that everything is working as expected. I can see all the endpoints but not working. To help you how to proceed, I suggest you refer this **[artlicle](https://www.theodinproject.com/lessons/ruby-on-rails-apis-and-building-your-own)** or **this one**. **Please make sure you taste everything is working before submitting a new review**